{
  try {
    go();
  }
 catch (  Throwable t) {
    internalHandleException(new IOException("Thread '" + getName() + "' terminated due to an exception: "+ t.getMessage(),t));
  }
}
